CRM/Core/Smarty/plugins/modifier.crmCountCharacters.php

```diff
@@ -0,0 +1,32 @@
+<?php
+/**
+ * Smarty plugin
+ * @package Smarty
+ * @subpackage plugins
+ */
+
+/**
+ * Smarty count_characters modifier plugin
+ *
+ * Type:     modifier<br>
+ * Name:     crmCountCharacteres<br>
+ * Purpose:  count the number of characters in a text with handling for NULL values
+ * @link http://smarty.php.net/manual/en/language.modifier.count.characters.php
+ *          count_characters (Smarty online manual)
+ * @author   Monte Ohrt <monte at ohrt dot com>
+ * @param string $string
+ * @param boolean $include_spaces include whitespace in the character count
+ * @return integer
+ */
+function smarty_modifier_crmCountCharacters($string, $include_spaces = FALSE) {
+  if ($include_spaces) {
+    return(strlen($string));
+  }
+
+  if (is_null($string)) {
+    return 0;
+  }
```

Comment:
I think this is mergeable as-is just noting some possible improvements:

* The check for null here could come before potentially passing to strlen()
* While copied from the original function, strlen isn't multibyte-friendly. In the places where it's used currently it doesn't make too much difference.
* The use in schema.xml is really checking for something like `if $field.default !== null && $field.default !== ''` (i.e. '0' is allowed but not null or blank). So could avoid the function.